Add pages for new documentation #25 (#26)
metafacture/metafacture-documentation: Home.md

![logo](https://github.com/culturegraph/metafacture-core/wiki/img/metafacture_small.png)

# Metafacture Documentation

> Metafacture is a toolkit for processing semi-structured data with a focus on library metadata. It provides a versatile set of tools for reading, writing and transforming data. Metafacture can be used as a stand-alone application via CLI or as a Java library in other applications. There is also a playground where you can test workflows.

Author: +1

The name Metafacture is a portmanteau of the words metadata and manufacture.

> Metafacture comprises three main parts: Framework, Flux and the Transformation-Module Fix. It can be extended with modules.

Reviewer: ```suggestion
Metafacture comprises three main parts: Framework, Flux and the Transformation modules Fix and Morph. It can be extended with modules.
```

Reviewer: Agreed.You would mention Morph without having to go into details - as you mention modules without going into details. I propose to a) remove Fix  in "Transformation module Fix" or b) mention them both. I vote for b) also because in the Playground while you encounter the textarea named "transformationFile" there may be written "metamorph" in it  ["Morph" https://github.com/metafacture/metafacture-playground/issues/114](https://test.metafacture.org/playground/?example=Local+formeta+to+XML+%28morph%29).

__________________

## Using Metafacture via playground or CLI

> [!NOTE]
> No JAVA-Code is necessary!!!

> While working with the playground or the command line you only need [Flux](#flux) and the transformation module [Fix](#fix).

Author: +1

Have a look here for [Getting started](/Getting-Started.md).

> ## Framework for JAVA-Integration/Development

Reviewer: ```suggestion
## Framework for Java integration/development
```

Author: +1

> If you plan to use Metafacture as a Java library or if you wish to add commands to Flux. You should get familar with the [Framework](/Home.md#framework).

Reviewer: ```suggestion
```

Reviewer: This was written before in the section describing using metafacture as a java library.

Author: not sure what you mean here?

Reviewer: It is not written _before_ but _later_ in this section:
In this file `Home.md` you have two times the sentence 
see https://github.com/metafacture/metafacture-documentation/blob/a096b54186d6502cd8881426dc21ed5125490ff0/Home.md#framework-for-java-integrationdevelopment
and see https://github.com/metafacture/metafacture-documentation/blob/a096b54186d6502cd8881426dc21ed5125490ff0/Home.md#framework.

Author: -1 for now i would keep it. Lets have a look when it is productive

__________________

## FLUX

> Flux is a scripting language to easily build and run processing pipelines. No Java programming is necessary, just a command line. To use Flux you may download the binary distribution of Metafacture.

Reviewer: ```suggestion
Flux is a scripting language to easily build and run processing pipelines. No Java programming is necessary - it's used as a command line. To use Flux you may download the binary distribution of Metafacture.
```

For more information on how to use Flux, see the [Flux User Guide](/Flux-User-Guide.md).

See [here for all available FLUX-Commands](/flux-commands.md).

## FIX

Metafix is a domain specific language for metadata transformation based on Catmandu FIX. The FIX object performing the transformation is used as part of a processing pipeline.

If you are using **Metafacture with CLI or Playground** and therefore the Flux scripting language to build and run pipelines, use the `fix` command in your FLUX-Pipeline. 

If you are using **Metafacture as a Java library**, just create a Metafix object and add it to your pipeline (see also the [Framework User Guide](#framework)).

The transformation itself is declared in a fix-object which can be a file. For more information on how to declare transformations see [Metafix User Guide](/Fix-User-Guide.md).

> [!NOTE]

> > PS: There is also the transformation modul MORPH but for that have a look at[ the old documentation](https://github.com/metafacture/metafacture-core/wiki/Metamorph-User-Guide) and the german cookbook by [Swissbib](https://swissbib.gitlab.io/metamorph-doku/).

Reviewer: ```suggestion
> PS: There is also the transformation modul MORPH. Have a look at[ the old documentation](https://github.com/metafacture/metafacture-core/wiki/Metamorph-User-Guide) and the german cookbook by [Swissbib](https://swissbib.gitlab.io/metamorph-doku/).
```
